pkg/instancetype/preference/apply/disk.go

```diff
@@ -0,0 +1,69 @@
+//nolint:gocyclo
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+package apply
+
+import (
+	virtv1 "kubevirt.io/api/core/v1"
+	v1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+)
+
+func applyDiskPreferences(preferenceSpec *v1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) {
+	for diskIndex := range vmiSpec.Domain.Devices.Disks {
+		vmiDisk := &vmiSpec.Domain.Devices.Disks[diskIndex]
+		// If we don't have a target device defined default to a DiskTarget so we can apply preferences
+		if vmiDisk.DiskDevice.Disk == nil && vmiDisk.DiskDevice.CDRom == nil && vmiDisk.DiskDevice.LUN == nil {
+			vmiDisk.DiskDevice.Disk = &virtv1.DiskTarget{}
+		}
+
+		if vmiDisk.DiskDevice.Disk != nil {
```

Comment:
nit: Could this be a switch-case?

Reply:
https://github.com/kubevirt/kubevirt/issues/13044